rewrite-java/src/main/java/org/openrewrite/java/cleanup/ReplaceRedundantFormatWithPrintf.java

```diff
@@ -0,0 +1,166 @@
+
+import java.util.List;
+
+public class ReplaceRedundantFormatWithPrintf extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "Replace redundant String format invocations that are wrapped with PrintStream operations";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Replaces `PrintStream.print(String.format(format, ...args))` with `PrintStream.printf(format, ...args)` (and for `println`, appends a newline to the format string).";
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new JavaIsoVisitor<ExecutionContext>() {
+            @Override
+            public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+                method = super.visitMethodInvocation(method, ctx);
+
+                if (!isPrintStreamPrintOrPrintln(method.getMethodType())) {
+                    return method;
+                }
+
+                if (method.getArguments().size() != 1) {
+                    return method;
+                }
+
+                Expression arg = method.getArguments().get(0);
+                if (!(arg instanceof J.MethodInvocation)) {
+                    return method;
+                }
+
+                J.MethodInvocation innerMethodInvocation = (J.MethodInvocation) arg;
+
+                if (!isStringFormat(innerMethodInvocation.getMethodType())) {
+                    return method;
+                }
+
+                List<Expression> newArguments = innerMethodInvocation.getArguments();
+
+                boolean needsNewline = method.getMethodType().getName().equals("println");
+                if (needsNewline) {
+                    Integer formatStringIndex = getFormatStringIndex(innerMethodInvocation.getMethodType());
+                    if (formatStringIndex == null) {
+                        return method;
+                    }
+
+                    Expression formatStringExpression = newArguments.get(formatStringIndex);
+                    if (!(formatStringExpression instanceof J.Literal)) {
+                        return method;
+                    }
+
+                    J.Literal formatStringLiteral = (J.Literal) formatStringExpression;
+                    Object formatStringValue = formatStringLiteral.getValue();
+                    if (!(formatStringValue instanceof String)) {
+                        return method;
+                    }
+
+                    formatStringLiteral = appendToStringLiteral(formatStringLiteral, "%n");
+                    newArguments = ListUtils.concatAll(
+                            ListUtils.concat(newArguments.subList(0, formatStringIndex), formatStringLiteral),
+                            newArguments.subList(formatStringIndex + 1, newArguments.size())
+                    );
+                }
+
+                return method.withName(method.getName().withSimpleName("printf")).withArguments(newArguments);
```

Comment:
Better to use `JavaTemplate` replacement here so that the new method receives the correct type attribution information as well.